spec.html

```diff
@@ -22683,6 +22683,21 @@
         </emu-alg>
       </emu-clause>
 
+      <emu-clause id="sec-object.getownpropertydescriptors">
+        <h1>Object.getOwnPropertyDescriptors ( _O_ )</h1>
+        <p>When the `getOwnPropertyDescriptors` function is called, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _obj_ be ? ToObject(_O_).
+          1. Let _ownKeys_ be ? _obj_.[[OwnPropertyKeys]]().
+          1. Let _descriptors_ be ! ObjectCreate(%ObjectPrototype%).
+          1. Repeat, for each element _key_ of _ownKeys_ in List order,
+            1. Let _desc_ be ? _obj_.[[GetOwnProperty]](_key_).
+            1. Let _descriptor_ be ! FromPropertyDescriptor(_desc_).
+            1. Perform ! CreateDataProperty(_descriptors_, _key_, _descriptor_).
```

Comment:
what happens if you `Object.getOwnpropertyDescriptor(proxy, key)` and/or
`Reflect.ownKeys(proxyObject)` ?

On Wed, May 25, 2016 at 3:24 PM, Claude Pache <notifications@github.com>
wrote:

> In spec.html
> <https://github.com/tc39/ecma262/pull/582#discussion_r64581739>:
>
> > @@ -22683,6 +22683,21 @@
> >          </emu-alg>
> >        </emu-clause>
> >
> > +      <emu-clause id="sec-object.getownpropertydescriptors">
> > +        <h1>Object.getOwnPropertyDescriptors ( _O_ )</h1>
> > +        <p>When the `getOwnPropertyDescriptors` function is called, the following steps are taken:</p>
> > +        <emu-alg>
> > +          1. Let _obj_ be ? ToObject(_O_).
> > +          1. Let _ownKeys_ be ? _obj_.[[OwnPropertyKeys]]().
> > +          1. Let _descriptors_ be ! ObjectCreate(%ObjectPrototype%).
> > +          1. Repeat, for each element _key_ of _ownKeys_ in List order,
> > +            1. Let _desc_ be ? _obj_.[[GetOwnProperty]](_key_).
> > +            1. Let _descriptor_ be ! FromPropertyDescriptor(_desc_).
> > +            1. Perform ! CreateDataProperty(_descriptors_, _key_, _descriptor_).
>
> Should we run this step also when *descriptor* is *undefined*? (Could
> happen if *obj* is a proxy, for instance.)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/tc39/ecma262/pull/582/files/55ac18633a0e68ba24b1f62288e3c77ddece58bd#r64581739>
>